src/backend/renderer/element/memory.rs

```diff
@@ -231,16 +246,20 @@ impl Default for MemoryRenderBufferInner {
 
 impl MemoryRenderBufferInner {
     fn new(
+        format: Fourcc,
         size: impl Into<Size<i32, Buffer>>,
         scale: i32,
         transform: Transform,
         opaque_regions: Option<Vec<Rectangle<i32, Buffer>>>,
     ) -> Self {
-        // TODO: Allow to specify the format when ImportMem
-        // has been changed to allow specifying a format
         let size = size.into();
         MemoryRenderBufferInner {
-            mem: vec![0; (size.w * 4 * size.h) as usize],
+            mem: vec![
+                0;
+                (size.w * (get_bpp(format).expect("Format with unknown bits per pixel") as i32) * size.h)
```

Comment:
Do we expect that `expect` to ever be invoked? I am not sure if returning a `Result` would be justified, but at least we should add a comment here and on `from_memory` warning about that panic if it can happen.

Reply:
A result only makes sense if there is a way to handle this case at runtime. Given we strive to implement `get_bpp` for every possible format and the `MemoryRenderElement` is meant to be used for software rendered content, where you usually know the format during compile time, it did occur to me that there would be reason to do that. But I guess I could convert this into a proper error case?